inc/commondbtm.class.php

```diff
@@ -2003,56 +2009,65 @@ function showDates($options = array()) {
          $colspan = 1;
       }
 
-      echo "<tr class='tab_bg_1 footerRow'>";
+      $out .= "<tr class='tab_bg_1 footerRow'>";
       //Display when it's not a new asset being created
       if ($date_creation_exists
          && $this->getID() > 0
             && (!isset($options['withtemplate']) || $options['withtemplate'] == 0)) {
-         echo "<th colspan='$colspan'>";
-         printf(__('Created on %s'), Html::convDateTime($this->fields["date_creation"]));
-         echo "</th>";
+         $out .= "<th colspan='$colspan'>";
+         $out .= sprintf(__('Created on %s'), Html::convDateTime($this->fields["date_creation"]));
+         $out .= "</th>";
       } else if (!isset($options['withtemplate']) || $options['withtemplate'] == 0) {
-         echo "<th colspan='$colspan'>";
-         echo "</th>";
+         $out .= "<th colspan='$colspan'>";
+         $out .= "</th>";
       }
 
       if (isset($options['withtemplate']) && $options['withtemplate']) {
-         echo "<th colspan='$colspan'>";
+         $out .= "<th colspan='$colspan'>";
          //TRANS: %s is the datetime of insertion
-         printf(__('Created on %s'), Html::convDateTime($_SESSION["glpi_currenttime"]));
-         echo "</th>";
+         $out .= sprintf(__('Created on %s'), Html::convDateTime($_SESSION["glpi_currenttime"]));
+         $out .= "</th>";
       }
 
       if ($date_mod_exists) {
-         echo "<th colspan='$colspan'>";
+         $out .= "<th colspan='$colspan'>";
          //TRANS: %s is the datetime of update
-         printf(__('Last update on %s'), Html::convDateTime($this->fields["date_mod"]));
-         echo "</th>";
+         $out .= sprintf(__('Last update on %s'), Html::convDateTime($this->fields["date_mod"]));
+         $out .= "</th>";
       } else {
-         echo "<th colspan='$colspan'>";
-         echo "</th>";
+         $out .= "<th colspan='$colspan'>";
+         $out .= "</th>";
       }
 
       if ((!isset($options['withtemplate']) || ($options['withtemplate'] == 0))
           && !empty($this->fields['template_name'])) {
-         echo "<th colspan='".($colspan * 2)."'>";
-         printf(__('Created from the template %s'), $this->fields['template_name']);
-         echo "</th>";
+         $out .= "<th colspan='".($colspan * 2)."'>";
+         $out .= sprintf(__('Created from the template %s'), $this->fields['template_name']);
+         $out .= "</th>";
       }
 
-      echo "</tr>";
+      $out .= "</tr>";
+
+      if ($options['display'] === true) {
```

Comment:
No, because `showDates()` is called from another method... Even if `GLPI_DISPLAY` is set to `ture`, the caller will explicitely set `$options['display']` to false and manage output itself (see https://github.com/glpi-project/glpi/pull/1632/files/ccc4ef31c8ae9eefc0aeb8680e5b4cb9f145fde6#diff-7f569484c496f62a6ccf54647fed2b1bR2107 as example).